3/assets/3.3-finished-code.rs

```diff
@@ -118,7 +118,12 @@ decl_module! {
 
             <balances::Module<T>>::make_transfer(&sender, &owner, kitty_price)?;
 
-            Self::transfer_from(owner.clone(), sender.clone(), kitty_id)?;
+            Self::transfer_from(owner.clone(), sender.clone(), kitty_id)
+                .expect("`owner` is shown to own the kitty; \
+                `owner` must have greater than 0 kitties, so transfer cannot cause underflow; \
+                `all_kitty_count` shares the same type as `owned_kitty_count`, \
```

Comment:
```suggestion
                `all_kitty_count` shares the same type as `owned_kitty_count` \
                and minting ensure there won't ever be more than `max()` kitties, \ 
```

Adding the mint proof, because without that there can still be an overflow...